src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -832,10 +840,7 @@ void MFEMSidreDataCollection::SetMesh(MPI_Comm comm, Mesh* new_mesh)
 {
   // use MFEMSidreDataCollection's custom SetMesh, then set MPI info
   SetMesh(new_mesh);
-
-  m_comm = comm;
-  MPI_Comm_rank(comm, &myid);
-  MPI_Comm_size(comm, &num_procs);
+  SetComm(comm);
```

Comment:
Please not that this changes the behavior, since SetComm also sets  serial = false;
  appendRankToFileName = true; (in addition to setting comm and num_procs and myid as done here). I believe that my change gives us the desired behavior and makes things consistent.

Reply:
How does `appendRankToFileName` interact with the new functionality that modifies the number of output files?  Please verify that this works as expected.

Reply:
I'm not familiar with how MFEM uses this symbol, but if it affects the string passed as the third argument in writer.write(), it may cause some trouble.  That string should be identical across all ranks making the call (which is something we should document better in axom).  SPIO will append an integer onto the output filenames, but it will be distinct from the rank number when `num_files < num_ranks`.

Reply:
I just added two tests, and played with it. With the change as indicated here, I see the correct output:

rzgenie2{taller1}69: ls dc_save_single_file/
dc_save_single_file_0000000.hdf5

rzgenie2{taller1}70: ls dc_save_two_files/
dc_save_two_files_0000000.hdf5	dc_save_two_files_0000001.hdf5

If I go back to the previous code, I see the same thing. So it seems that this change is ok, but I could switch it back if you want.

Reply:
Yes, those are the filenames I would expect, with the integers appended by SPIO.  Looks like `appendRankToFileName` isn't affecting anything that is passed to SPIO, and maybe it isn't doing anything at all.  The numbers are going to be `0` to `num_files-1` and will only have a direct mapping to the rank numbers in the `num_files == num_ranks` case.